ansible/roles/disable_hyperthreading/tasks/main.yml

```diff
@@ -0,0 +1,32 @@
+---
+- name: disable hyper-threading
+  block:
+  - name: install disableHT.sh
+    copy:
+      src: files/disableHT.sh
+      dest: /usr/local/bin/disableHT.sh
+      owner: root
+      group: root
+      mode: 0755
+    register: dh_script_out
+  - name: install disableHT.service
+    copy:
+      src: files/disableHT.service
+      dest: /etc/systemd/system/disableHT.service
+      owner: root
+      group: root
+      mode: 0644
+    register: dh_service_out
+  - name: enable disableHT.service
+    systemd:
+      name: disableHT
+      enabled: yes
+  - name: start disableHT.service (if something changed)
+    systemd:
+      name: disableHT
+      state: started
+      daemon_reload: yes
+    when: dh_service_out.changed or dh_script_out.changed
```

Comment:
Ansible is changing a little bit how conditions like "changed", "skipped" or "succeeded" are written: https://docs.ansible.com/ansible/2.6/user_guide/playbooks_conditionals.html#the-when-statement

I think moving forward, the way to write this would be ``dh_service_out is changed or dh_script_out is changed``.
You could also do away with this entirely by using a handler instead which is automatically fired when the task is changed so you don't need to look at the conditions for idempotency.
So, example:
```
  - name: install disableHT.service
    copy:
      src: files/disableHT.service
      dest: /etc/systemd/system/disableHT.service
      owner: root
      group: root
      mode: 0644
    notify:
      - reload disable hyperthread
```

and then a task in ``handlers/main.yaml`` with that name that does what you want.